smu/geometry/utilities.py

```diff
@@ -293,3 +323,83 @@ def is_single_fragment(bond_topology):
   number_visited = np.count_nonzero(visited) + visit(
       attached, a_multiply_connected_atom, visited)
   return number_visited == natoms
+
+def geom_to_angstroms(geometry:dataset_pb2.Geometry) -> dataset_pb2.Geometry:
+  """Convert all the coordinates in `geometry` to Angstroms
+  Args:
+    geometry: starting Geometry
+  Returns
+    New Geometry with adjusted coordinates.
+  """
+  result = dataset_pb2.Geometry()
+  for atom in geometry.atom_positions:
+    new_atom = dataset_pb2.Geometry.AtomPos()
+    new_atom.x = smu_utils_lib.bohr_to_angstroms(atom.x)
+    new_atom.y = smu_utils_lib.bohr_to_angstroms(atom.y)
+    new_atom.z = smu_utils_lib.bohr_to_angstroms(atom.z)
+    result.atom_positions.append(new_atom)
+
+  return result
+
+def max_bonds_any_form(atype: dataset_pb2.BondTopology.AtomType) -> int:
+  """Return the max number of bonds for any form of `atype`.
+
+    Args:
+      atype: a dataset_pb2 atom type
+    Returns:
+      max number of bonds
+  """
+  if atype in [dataset_pb2.BondTopology.ATOM_C,
+               dataset_pb2.BondTopology.ATOM_NPOS,
+               dataset_pb2.BondTopology.ATOM_O,
+               dataset_pb2.BondTopology.ATOM_F,
+               dataset_pb2.BondTopology.ATOM_H]:
+    return smu_utils_lib.ATOM_TYPE_TO_MAX_BONDS[atype]
+
+  if atype == dataset_pb2.BondTopology.ATOM_N:
+    return 4
+
+  if atype == dataset_pb2.BondTopology.ATOM_ONEG:
+    return 2
+
+def convert_to_neutral(bt: dataset_pb2.BondTopology) -> bool:
```

Comment:
You have a very similar function in smu_molecule.   Is there a meaningful difference?

Reply:
I think they are slightly different, of at least different enough that trying to merge them would look messy...